BO_training/mixed_test_func/scallop_casmopolitan_yaml.py

```diff
@@ -56,14 +56,15 @@ def Scallop_base(index,x,Result,input_file,software_path,docs):
 
     #NOTE this chr part may be removed
     #check if output gft is started with chr, if remove it
-    cmd = docs['precheck']['check_command'].format(pid)
-    chr_header = int(subprocess.getoutput(cmd))
-    print(f'number of transcript start with chr: {chr_header}')
-    #NOTE: remove 'chr' if ref genome doesn't start with chr
-    if chr_header>0:
-        cmd = docs['precheck']['excute_command'].format(pid)
-        print(cmd)
-        os.system(cmd)
+    if precheck == True:
+        cmd = docs['precheck']['check_command'].format(pid)
```

Comment:
I would not create that `precheck` option. Instead, detect whether or not it is given in the YAML file:

```Python
if docs.get('precheck') is not None and docks['precheck'].get('check_command') is not None:
  cmd = docks['precheck']['check_command'].format(pid)
  ...
```